optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/ScoreDirectorFactoryFactory.java

```diff
@@ -95,18 +77,29 @@ public ScoreDirectorFactoryFactory(ScoreDirectorFactoryConfig config) {
 
     protected AbstractScoreDirectorFactory<Solution_, Score_> decideMultipleScoreDirectorFactories(
             ClassLoader classLoader, SolutionDescriptor<Solution_> solutionDescriptor) {
-        EasyScoreDirectorFactory<Solution_, Score_> easyScoreDirectorFactory =
-                buildEasyScoreDirectorFactory(solutionDescriptor);
-        AbstractConstraintStreamScoreDirectorFactory<Solution_, Score_> constraintStreamScoreDirectorFactory =
-                buildConstraintStreamScoreDirectorFactory(solutionDescriptor);
-        IncrementalScoreDirectorFactory<Solution_, Score_> incrementalScoreDirectorFactory =
-                buildIncrementalScoreDirectorFactory(solutionDescriptor);
-        DroolsScoreDirectorFactory<Solution_, Score_> droolsScoreDirectorFactory = buildDroolsScoreDirectorFactory(
-                classLoader, solutionDescriptor);
+        // Load all known Score Director Factories via SPI.
+        ServiceLoader<ScoreDirectorFactoryProvider> scoreDirectorFactoryProviders =
```

Comment:
Code style: variable name is plural, but it's not an array
Can we rename it to scoreDirectorFactoryProviderLoader?